sae_lens/training/activations_store.py

```diff
@@ -451,27 +455,42 @@ def get_activations(self, batch_tokens: torch.Tensor):
 
         n_batches, n_context = batch_tokens.shape
 
-        stacked_activations = torch.zeros((n_batches, n_context, 1, self.d_in))
+        # For some models, we might want to exclude some positions from the sequence to train on
+        training_context_slice = list(
+            range(self.start_pos_offset, n_context - self.end_pos_offset)
+        )
+        # The size of the context window after slicing with position offsets
+        training_context_size = len(training_context_slice)
+
+        stacked_activations = torch.zeros(
+            (n_batches, training_context_size, 1, self.d_in)
+        )
 
         if self.hook_head_index is not None:
             stacked_activations[:, :, 0] = layerwise_activations[self.hook_name][
-                :, :, self.hook_head_index
+                :, training_context_slice, self.hook_head_index
             ]
         elif (
             layerwise_activations[self.hook_name].ndim > 3
         ):  # if we have a head dimension
             try:
                 stacked_activations[:, :, 0] = layerwise_activations[
                     self.hook_name
-                ].view(n_batches, n_context, -1)
+                ].view(n_batches, training_context_size, -1)[
```

Comment:
This looks like it's not correct - don't we want to first stack into `n_context` sequences and then lop off the start and end? It looks like as this is written it will turn this into `training_context_size` sequences and then further slice them, likely leading to the wrong size sequences. Is this covered by a test?